external/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisInputDStream.scala

```diff
@@ -249,6 +252,17 @@ object KinesisInputDStream {
     }
 
     /**
+      * Sets the [[SparkAWSCredentials]] to use for authenticating to the AWS CloudWatch
+      * endpoint. Will use the same credentials used for AWS Kinesis if no custom value is set.
+      *
+      * @param conf: Map[String, String] to use for CloudWatch authentication
+      */
+    def kinesisConf(conf: Map[String, String]): Builder = {
```

Comment:
If using ```SparkConf``` to store these custom config values doesn't end up being feasible then I'd strongly prefer that we follow the existing approach and have separate builder methods for setting the retry wait time and max attempts.